misk-hibernate/src/test/kotlin/misk/hibernate/CidGenerator.kt

```diff
@@ -0,0 +1,165 @@
+@file:JvmName("CidGenerator")
+
+package misk.hibernate
+
+import org.hibernate.HibernateException
+import org.hibernate.MappingException
+import org.hibernate.dialect.Dialect
+import org.hibernate.engine.jdbc.spi.JdbcServices
+import org.hibernate.engine.spi.SessionImplementor
+import org.hibernate.engine.spi.SharedSessionContractImplementor
+import org.hibernate.id.AbstractPostInsertGenerator
+import org.hibernate.id.Configurable
+import org.hibernate.id.PostInsertIdentityPersister
+import org.hibernate.id.insert.Binder
+import org.hibernate.id.insert.IdentifierGeneratingInsert
+import org.hibernate.id.insert.InsertGeneratedIdentifierDelegate
+import org.hibernate.persister.entity.SingleTableEntityPersister
+import org.hibernate.pretty.MessageHelper
+import org.hibernate.service.ServiceRegistry
+import org.hibernate.sql.Insert
+import org.hibernate.type.Type
+import java.io.Serializable
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.SQLException
+import java.util.Properties
+
+@Suppress("RedundantVisibilityModifier", "unused")
+public class CidGenerator : AbstractPostInsertGenerator(), Configurable {
+  private lateinit var parentColumn: String
+
+  @Throws(MappingException::class)
+  override fun configure(type: Type, params: Properties, serviceRegistry: ServiceRegistry) {
+    this.parentColumn = params.getProperty("parentColumn")
+    checkNotNull(parentColumn)
+  }
+
+  fun generate(s: SessionImplementor, obj: Any): Serializable {
```

Comment:
curious – why override?

Reply:
I blame the IntelliJ Java->Kotlin converter.